include/xtensor/xsort.hpp

```diff
@@ -144,6 +144,163 @@ namespace xt
+            using type = xarray<VT, L>;
+        };
+
+        template <class VT, class EC, size_t N, layout_type L>
+        struct rebind_value_type<VT, xtensor<EC, N, L>>
+        {
+            using type = xtensor<VT, N, L>;
+        };
+
+        template <class VT, class ET, class S, layout_type L>
+        struct rebind_value_type<VT, xtensor_fixed<ET, S, L>>
+        {
+            using type = xtensor_fixed<VT, S, L>;
+        };
+
+        template <class T>
+        struct argsort_result_type
+        {
+            using type = typename rebind_value_type<typename T::temporary_type::size_type,
+                                                    typename T::temporary_type>::type;
+        };
+
+        template <class Ed, class Ei>
+        void argsort_over_leading_axis(const Ed& data, Ei& inds)
+        {
+            using value_type = typename Ed::value_type;
+            std::size_t n_iters = 1;
+            std::ptrdiff_t data_secondary_stride;
+            std::ptrdiff_t inds_secondary_stride;
+            if (data.layout() == layout_type::row_major)
+            {
+                n_iters = std::accumulate(data.shape().begin(), data.shape().end() - 1,
+                                          std::size_t(1), std::multiplies<>());
+                data_secondary_stride = static_cast<std::ptrdiff_t>(data.strides()[data.dimension() - 2]);
+                inds_secondary_stride = static_cast<std::ptrdiff_t>(inds.strides()[inds.dimension() - 2]);
+            }
+            else
+            {
+                n_iters = std::accumulate(data.shape().begin() + 1, data.shape().end(),
+                                          std::size_t(1), std::multiplies<>());
+                data_secondary_stride = static_cast<std::ptrdiff_t>(data.strides()[1]);
+                inds_secondary_stride = static_cast<std::ptrdiff_t>(inds.strides()[1]);
+            }
+
+            std::ptrdiff_t data_offset = 0;
+            std::ptrdiff_t inds_offset = 0;
+
+            for (std::size_t i = 0; i < n_iters; ++i, data_offset += data_secondary_stride,
+                    inds_offset += inds_secondary_stride)
+            {
+                auto comp = [&data, &data_offset](std::size_t x, std::size_t y) {
+                    return (*(data.data() + data_offset + x) <
+                            *(data.data() + data_offset + y));
+                };
+                std::iota(inds.data() + inds_offset, inds.data() + inds_offset + inds_secondary_stride, 0);
+                std::sort(inds.data() + inds_offset, inds.data() + inds_offset + inds_secondary_stride, comp);
+            }
+        }
+    }
+
+    template <class E>
+    auto argsort(const xexpression<E>& e, placeholders::xtuph /*t*/)
+    {
+        using value_type = typename E::value_type;
+        using result_type = typename detail::argsort_result_type<E>::type;
+
+        const auto& de = e.derived_cast();
+
+        result_type result;
+        result.resize({de.size()});
```

Comment:
Resize does not do anything with `xtensor_fixed`. Maybe a good way to do this would be to use the constructor, and pass `{de.size()}`.

Reply:
Correction: I think the current version is fine (discussed in person with Sylvain). The suggestion would lead to a couple of other problems with xtensor_fixed :)